needy/cache.py

```diff
@@ -120,13 +120,14 @@ def lease_file(self, key, timeout=None, **kwargs):
         directory, and then save modifications when the lease is returned. If
         the path is removed during the lease, the key will be unset.'''
         with self.lease(key, timeout=timeout, **kwargs):
-            with tempfile.NamedTemporaryFile() as f:
```

Comment:
`NamedTemporaryFile` doesn't seem to behave well on Windows. We're wanting to effectively get a name for a path that we can read, write, delete, or create a file at as long as we're in the context. `NamedTemporaryFile` doesn't work for that sort of thing, so I'm replacing it with `TempDir` in a few places.